mage.ts

```diff
@@ -2415,7 +2415,7 @@ declare namespace mage {
              *
              * @memberOf State
              */
-            emit<T>(actorId: string | string[], eventName: string | number, data: T): void;
+            emit<T>(actorId: string | string[], eventName: string | number, data: T, configuration: mage.core.IStateEmitConfig): void;
```

Comment:
I can see cases where this would not behave as intended, since myEvents are currently returned only on close.

What I would suggest is that we create a separate method that will:

  1. Push myEvents into the rest of the event list
  2. Then call emitEvents

@ronkorving would that make sense to you?